build(deps): Bump Pydantic from v1 to v2

gavindsouza opened the pull request.

gavindsouza committed:
build(deps): Bump Pydantic from v1 to v2

gavindsouza committed:
test: Add tests for validate_argument_types decorator

gavindsouza commented:
Bumped to Pydantic `v2.0b2` for now, until the stable version is released.

ankush commented:
@gavindsouza lets wait for a stable release? No really urgent. 


`develop` runs in prod and is almost equivalent to v15 "pre-release".

gavindsouza commented:
I wasn't proposing merging this with Pydantic beta. Stable release gonna roll out in a couple of weeks - so this is just a run to make sure things aren't too breaking.

gavindsouza committed:
test: Add tests for Document arg validations

gavindsouza committed:
fix: Allow arbitrary objects validation in TypeAdapter

gavindsouza committed:
Merge branch 'develop' into bump-pydantic-v2

gavindsouza commented:
/remind me about this in a week

ankush commented:
Ref

https://docs.pydantic.dev/latest/blog/pydantic-v2/

gavindsouza committed:
Merge branch 'develop' into bump-pydantic-v2

gavindsouza commented:
Bumped to Pydantic v2.0b3 with the resolved merge conflict

gavindsouza committed:
build: Bump to stable Pydantic v2

gavindsouza committed:
Merge branch 'develop' into bump-pydantic-v2

gavindsouza committed:
perf: Defer pydantic imports until function call

Pydantic adds an additional 1-2MB in memory usage. We can defer it in
case an environment doesn't use it at all.

gavindsouza committed:
test: Bump memory usage treshold post pydantic v2 upgrade

ankush committed:
perf: preload pydantic

ankush committed:
Revert "test: Bump memory usage treshold post pydantic v2 upgrade"

This reverts commit af7032893087db31f5eff6c31c9473561fdd4d0b.
